### ISSUE TITLE
overload getindex to allow PopData filtering like a normal dataframe

### ISSUE THREAD

**Issue by pdimens** [BioJulia/PopGenIO.jl#14]
**Is your feature request related to a problem and which?**
The current inability to filter PopData naturally, i.e. with `getindex` stuff

**Describe the solution/feature you'd like (with examples)**
```julia
x = @nancycats

subset = x[(x.genodata.name .!= "N102") .& (x.genodata.locus .!= "fca8") .& (x.genodata.population .!= "3"), :]
PopData{Diploid, 8 Microsatellite loci}
  Samples: 225
  Populations: 16
```

**additional info**
What needs to happen specifically is to allow `getindex` to work on a `PopData` and just ignore the `metadata` field, then update the `metadata` with the current samples (if any changed) to spit out a new `PopData`

**Comment by pdimens** [comment]
commit https://github.com/BioJulia/PopGenCore.jl/commit/601a5b6b096470ab6250677a1084553f95e2b4ba starts addressing this

**Comment by pdimens** [comment]
Added overloaded `filter` and `filter!` with https://github.com/BioJulia/PopGenCore.jl/commit/afc5deba1ef4af7225ae4da62e84ff622647b8f0 
- This now expands the use of those `DataFrames.jl` functions on `PopData` directly, with the only difference being that the "haystack" (the PopData) is the first argument and the "needle" is the second
- This also simplifies the logic for `exclude!`
- Overloaded `filter!` now lets `keep!` to work on multiple fields!

**Comment by pdimens** [comment]
fixed with release of `0.2.0`
